pkgs/applications/misc/zola/default.nix

```diff
@@ -13,9 +24,18 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "1hg8j9a8c6c3ap24jd96y07rlp4f0s2mkyx5034nlnkm3lj4q42n";
 
-  nativeBuildInputs = [ cmake pkg-config installShellFiles];
-  buildInputs = [ openssl oniguruma ]
-    ++ lib.optional stdenv.isDarwin CoreServices;
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    installShellFiles
+  ];
+  buildInputs = [
+    openssl
+    oniguruma
+  ] ++ lib.optional stdenv.isDarwin [
```

Comment:
``lib.optionals``

Reply:
Since `optional` acts on a single item as opposed to a list, it would produce a `buildInputs` of `[ [ ... ] ]`. Isn't this invalid?

How does this even build in the first place, if so?

Reply:
> Isn't this invalid?

I am not sure what actually happens but unless you eval on darwin we don't know. Maybe it is flattened at some point or eval fails on darwin entirely.